packages/create-docusaurus/src/index.ts

```diff
@@ -17,13 +16,40 @@ import supportsColor from 'supports-color';
 const RecommendedTemplate = 'classic';
 const TypeScriptTemplateSuffix = '-typescript';
 
-function hasYarn() {
-  try {
-    execSync('yarnpkg --version', {stdio: 'ignore'});
-    return true;
-  } catch (e) {
-    return false;
+const DefaultPackageManager = 'npm';
+const SupportedPackageManagers = {
+  npm: 'package-lock.json',
+  yarn: 'yarn.lock',
+  pnpm: 'pnpm-lock.yaml',
+};
+
+type SupportedPackageManager = keyof typeof SupportedPackageManagers;
+
+function getPackageManagerForUse(
+  forceUseNpm?: boolean,
+): SupportedPackageManager {
+  if (forceUseNpm) {
+    return 'npm';
+  }
+
+  const packageManagers = Object.keys(
+    SupportedPackageManagers,
+  ) as SupportedPackageManager[];
+  const packageManagerFromLockFile = packageManagers.find((packageManager) =>
+    fs.existsSync(
+      path.resolve(process.cwd(), SupportedPackageManagers[packageManager]),
+    ),
+  );
```

Comment:
Should we look up a few folders? In case the user is creating the website at `packages/website` and the lock file is at the monorepo's root.

Reply:
It is assumed that the starter command will be running from the root of project in any case. So I didn't want to complicate the process with this at the moment.

Reply:
Sure. `create-docusaurus` is not very well-documented now, there are some interesting features that few users know. I'll probably improve that later